### ISSUE TITLE
FR: support dot-repeat

### ISSUE THREAD

**Issue by chrisgrieser** [supermaven-inc/supermaven-nvim#55]
One feature that would be very useful is to dot-repeat (`.`) the previous completion made the AI. 

For reference, of all the AI completion plugins I tried, I think only [neocodeium](https://github.com/monkoose/neocodeium), has that feature.

**Comment by chrisgrieser** [comment]
> Basically storing last accepted suggestion and when pressing . rewrite it under the cursor?

exactly!

**Comment by AlejandroSuero** [comment]
@chrisgrieser I implemented the `dot-repeat` functionality in the PR linked here.

Personally, I wouldn't use `dot-repeat` with `.` as it conflicts with `.` functionality provided by Neovim itself, but maybe introducing a keymap like `<C-r>` for repeat in **Insert Mode** could be better as default, and exposing `repeat_suggestion` under `keymaps` to change the default one.

**Comment by sm-victorw** [comment]
We might revisit this later, but currently it feels a bit too niche of a feature to justify

**Comment by rodrigoscc** [comment]
@sm-victorw dot repeat is a core feature of neovim, it isn't a niche feature. Please reconsider.

**Comment by devenv** [comment]
I agree with @rodrigoscc, dot repeat is a core feature, please reconsider native support or tpope/vim-repeat.
Being unable to repeat the last action is very limiting.

**Comment by chrisgrieser** [comment]
I agree, dot-repeat is a core vim feature, it feels awkward not being able to dot-repeat.

**Comment by chrisgrieser** [comment]
@AlejandroSuero Considering you authored various useful PRs for supermaven that were never merged (#55, #49, …) and that there is little activity at this repo anymore, would you maybe consider maintaining a fork?

`supermaven-nvim` is MIT-licensed, so I assume the company will have no issue with it.

[neocodeium](http://github.com/monkoose/neocodeium) is also a lua plugin maintained by someone from the community rather than codeium themselves, because they wanted more features.

**Comment by AlejandroSuero** [comment]
@chrisgrieser I was considering it, I will try to speak to the maintainers in case they grant me access to this repo, this way would be easier instead of telling everyone to switch, and make changes from this repo.

I will be busy during the next 2 weeks, and if I receive no answer or they tell me to open another one instead, I will start after 2 weeks.
